src/__tests__/setup.ts

```
@@ -0,0 +1,425 @@
+}
+
+// Mock Git Service
+export interface MockGitService {
+  readonly isAvailable: Effect.Effect<boolean>
+  readonly clone: (url: string, target: string) => Effect.Effect<void, { _tag: 'ExecCommandError', command: string, cause: unknown }>
+  readonly pull: (directory: string) => Effect.Effect<void, { _tag: 'ExecCommandError', command: string, cause: unknown }>
+  readonly setAvailable: (available: boolean) => Effect.Effect<void>
+  readonly setCloneResult: (result: 'success' | 'failure') => Effect.Effect<void>
+}
+
+export class MockGitService extends Context.Tag('MockGitService')<
```

Biome (2.1.2)

[error] This class is unsafely merged with an interface.

The interface is declared here.

The TypeScript compiler doesn't check whether properties defined in the interface are initialized in the class.

(lint/suspicious/noUnsafeDeclarationMerging)